resources/functionMap.php

```diff
@@ -7529,9 +7529,9 @@
 'mysqli_result::fetch_all' => ['array', 'resulttype='=>'int'],
 'mysqli_result::fetch_array' => ['array|null', 'resulttype='=>'int'],
 'mysqli_result::fetch_assoc' => ['array<string,string>|null'],
-'mysqli_result::fetch_field' => ['object|false'],
-'mysqli_result::fetch_field_direct' => ['object|false', 'fieldnr'=>'int'],
-'mysqli_result::fetch_fields' => ['array'],
+'mysqli_result::fetch_field' => ['(stdClass&object{name: string, orgname: string, table: string, orgtable: string, def: string, db: string, catalog: "def", max_length: int, length: int, charsetnr: string, flags: int, type: int, decimals: int})|false'],
+'mysqli_result::fetch_field_direct' => ['(stdClass&object{name: string, orgname: string, table: string, orgtable: string, def: string, db: string, catalog: "def", max_length: int, length: int, charsetnr: string, flags: int, type: int, decimals: int})|false', 'fieldnr'=>'int'],
```

Comment:
Either the PHP docs are wrong, or this PR is wrong. Looking at this https://www.php.net/manual/en/mysqli-result.fetch-field-direct.php it's not exactly 1:1. For example `db` and `catalog` fields are missing in PHP docs. Can you please verify the return types are the same, and send a PR to PHP docs? Thanks.

Reply:
As far as I can tell based on the `php_add_field_properties` function linked in the PR description and by trying the 3 functions/methods myself: the PR is right, the docs are wrong.

```php
<?php

declare(strict_types=1);

$db = new mysqli(
	'127.0.0.1',
	'root',
	'root'
);
$result = $db->query('SELECT 1');
var_dump($result->fetch_fields());
var_dump($result->fetch_field());
var_dump($result->fetch_field_direct(0));

$result = $db->query('SELECT 1');
var_dump(mysqli_fetch_fields($result));
var_dump(mysqli_fetch_field($result));
var_dump(mysqli_fetch_field_direct($result, 0));
```

Gives the following result (PHP 8.3, MariaDB 10.11):

```
array(1) {
  [0]=>
  object(stdClass)#3 (13) {
    ["name"]=>
    string(1) "1"
    ["orgname"]=>
    string(0) ""
    ["table"]=>
    string(0) ""
    ["orgtable"]=>
    string(0) ""
    ["def"]=>
    string(0) ""
    ["db"]=>
    string(0) ""
    ["catalog"]=>
    string(3) "def"
    ["max_length"]=>
    int(0)
    ["length"]=>
    int(1)
    ["charsetnr"]=>
    int(63)
    ["flags"]=>
    int(32897)
    ["type"]=>
    int(3)
    ["decimals"]=>
    int(0)
  }
}
object(stdClass)#3 (13) {
  ["name"]=>
  string(1) "1"
  ["orgname"]=>
  string(0) ""
  ["table"]=>
  string(0) ""
  ["orgtable"]=>
  string(0) ""
  ["def"]=>
  string(0) ""
  ["db"]=>
  string(0) ""
  ["catalog"]=>
  string(3) "def"
  ["max_length"]=>
  int(0)
  ["length"]=>
  int(1)
  ["charsetnr"]=>
  int(63)
  ["flags"]=>
  int(32897)
  ["type"]=>
  int(3)
  ["decimals"]=>
  int(0)
}
object(stdClass)#3 (13) {
  ["name"]=>
  string(1) "1"
  ["orgname"]=>
  string(0) ""
  ["table"]=>
  string(0) ""
  ["orgtable"]=>
  string(0) ""
  ["def"]=>
  string(0) ""
  ["db"]=>
  string(0) ""
  ["catalog"]=>
  string(3) "def"
  ["max_length"]=>
  int(0)
  ["length"]=>
  int(1)
  ["charsetnr"]=>
  int(63)
  ["flags"]=>
  int(32897)
  ["type"]=>
  int(3)
  ["decimals"]=>
  int(0)
}
array(1) {
  [0]=>
  object(stdClass)#2 (13) {
    ["name"]=>
    string(1) "1"
    ["orgname"]=>
    string(0) ""
    ["table"]=>
    string(0) ""
    ["orgtable"]=>
    string(0) ""
    ["def"]=>
    string(0) ""
    ["db"]=>
    string(0) ""
    ["catalog"]=>
    string(3) "def"
    ["max_length"]=>
    int(0)
    ["length"]=>
    int(1)
    ["charsetnr"]=>
    int(63)
    ["flags"]=>
    int(32897)
    ["type"]=>
    int(3)
    ["decimals"]=>
    int(0)
  }
}
object(stdClass)#2 (13) {
  ["name"]=>
  string(1) "1"
  ["orgname"]=>
  string(0) ""
  ["table"]=>
  string(0) ""
  ["orgtable"]=>
  string(0) ""
  ["def"]=>
  string(0) ""
  ["db"]=>
  string(0) ""
  ["catalog"]=>
  string(3) "def"
  ["max_length"]=>
  int(0)
  ["length"]=>
  int(1)
  ["charsetnr"]=>
  int(63)
  ["flags"]=>
  int(32897)
  ["type"]=>
  int(3)
  ["decimals"]=>
  int(0)
}
object(stdClass)#2 (13) {
  ["name"]=>
  string(1) "1"
  ["orgname"]=>
  string(0) ""
  ["table"]=>
  string(0) ""
  ["orgtable"]=>
  string(0) ""
  ["def"]=>
  string(0) ""
  ["db"]=>
  string(0) ""
  ["catalog"]=>
  string(3) "def"
  ["max_length"]=>
  int(0)
  ["length"]=>
  int(1)
  ["charsetnr"]=>
  int(63)
  ["flags"]=>
  int(32897)
  ["type"]=>
  int(3)
  ["decimals"]=>
  int(0)
}
```
I will see about updating the PHP documentation, but it's not that simple because there are other inconsistencies and I don't have sufficient idea about the precise meaning of the fields.